ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/security/KsqlUserContextProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.security;
+
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+
+import io.confluent.ksql.rest.server.context.ConfiguredKafkaClientSupplier;
+import java.security.Principal;
+import java.util.function.Supplier;
+
+/**
+ * Provides access to clients required to communicate with remote services using the context of
+ * a specified user name and/or credentials.
+ * <p/>
+ * This context is used by KSQL to access Kafka and/or Schema Registry resources when a user
```

Comment:
Let's add a note similar to the one in `KsqlSecurityExtension#getUserContextProvider()` to call out that the user context is currently only used for non-persistent queries.

Reply:
Done